source/reference/data-types.txt

```diff
@@ -38,325 +40,495 @@ string or as a ``Date`` object:
+   myDateString
 
-#. Retrieve the updated document:
+The result is the value of ``myDateString``:
 
-   .. code-block:: javascript
+.. code-block:: javascript
+   :copyable: false
 
-      db.collection.findOne( { _id: 10 } )
+   Wed Dec 19 2012 01:03:25 GMT-0500 (EST)
 
-   In the updated document, the ``calc`` field contains a floating
-   point value:
+To verify the type, use the ``typeof`` operator, as in the following:
 
-   .. code-block:: sh
+.. code-block:: javascript
 
-      { "_id" : 10, "calc" : 2555555000010 }
+   typeof myDateString
 
-.. note::
+The operation returns ``string``.
 
-   In the legacy :binary:`~bin.mongo` shell, ``NumberLong`` can accept
-   either a string or integer value. In ``mongosh``, ``NumberLong`` can
-   only accept a string value.
+Return ``Date``
+~~~~~~~~~~~~~~~
 
-.. _shell-type-int:
+``mongosh`` wraps objects of ``Date`` type with the
+``ISODate`` helper; however, the objects remain of type ``Date``.
 
-NumberInt
-~~~~~~~~~
+The following example uses both the ``new Date()`` constructor and the
+``ISODate()`` constructor to return ``Date`` objects.
 
-If a number can be converted to a 32-bit ``int``, ``mongosh`` will
-store it in that form. If not, ``mongosh`` defaults to storing the
-number as a ``double``. The ``NumberInt()`` constructor can be used to
-explicitly specify 32-bit integers.
+.. code-block:: javascript
 
-.. _shell-type-decimal:
+   var myDate = new Date();
+   var myDateInitUsingISODateWrapper = ISODate();
 
-NumberDecimal
-~~~~~~~~~~~~~
+You can use the ``new`` operator with the ``ISODate()`` constructor as
+well.
 
-If a number can be converted to a 32-bit ``int``, ``mongosh`` will
-store it in that form. If not, ``mongosh`` defaults to storing the
-number as a ``double``. The ``NumberDecimal()`` constructor can be
-used to explicitly specify 128-bit decimal-based floating-point values
-that are capable of emulating decimal rounding with exact precision. 
+To print the value of the variable, type the variable name in the
+shell, as in the following:
 
-This functionality is intended for applications that handle
-:manual:`monetary data </tutorial/model-monetary-data>`, such as
-financial, tax, and scientific computations. 
+.. code-block:: javascript
 
-The ``decimal`` :manual:`BSON type </reference/bson-types>`
-uses the IEEE 754 decimal128 floating-point numbering format which
-supports 34 decimal digits (i.e. significant digits) and an exponent
-range of −6143 to +6144.
+   myDate
 
-The ``NumberDecimal()`` constructor accepts the ``decimal`` value as a
-string:
+The result is the ``Date`` value of ``myDate`` wrapped in the
+``ISODate()`` helper:
 
 .. code-block:: javascript
+   :copyable: false
 
-   NumberDecimal("1000.55")
-   
-The value is stored in the database as follows:
-
-.. code-block:: javascript
+   ISODate("2012-12-19T06:01:17.171Z")
 
-   NumberDecimal("1000.55")
+To verify the type, use the ``instanceof`` operator, as in the
+following:
```

Comment:
Ditto here – if you really **need** to verify that something is a Date, you can use `Object.prototype.toString.call(value) === [object Date]`

Reply:
Done